### ISSUE TITLE
Some details of linear evaluation protocol

### ISSUE THREAD

**Issue by Yingdong-Hu** [wvangansbeke/Unsupervised-Semantic-Segmentation#7]
Hi, @wvangansbeke 
I have two questions about the details of linear evaluation protocol.

1. When you train a 1*1 convolutional layer on top of MoCo v2 features, do you modify the stride of convolutional layer in the ResNet backbone ?  What's the final downsampling factor ?
2. Do all models have the same finetuning settings (learning rate, train time) ?

**Comment by vobecant** [comment]
Dear authors,

I am also interested in how do you deal with the different size of the feature maps that come from MoCo v2. Can you please provide the answer?

Thank you very much in advance.

**Comment by wvangansbeke** [comment]
Hi @Alxead @vobecant 

Apologies for the late reply. Thanks for the interest.

1. We used the ResNet-50 backbone with a dilation-rate of 8 both for MoCo and our DeepLab setup. We also tested the standard ResNet backbone for the linear evaluation protocol but took the ones with the best results. In order to match the resolutions we simply used bilinear interpolation.

2. Our settings can be found in the config files. They are all mostly the same when using saliency supervision or not. I want to point out that the optimal settings differ a lot between methods (e.g. SwAV, SeLa, SimCLR). For example SwAV uses a much larger learning rate for (linear) finetuning than MoCo. So there does not seem to be one set of parameters that works well for everything.

Hope this helps.

**Comment by vobecant** [comment]
Hi @wvangansbeke ,

thank you for your fast reply to my question. I have few more:

1. If I understand it correctly, in your setup, you train the ResNet-50 with a dilation rate of 8 with a DeepLab-like head attached to it and then remove the head during evaluation? 
2. What is the output resolution of your method with the attached head? Is it of the same size as the input image?

Thank you.

**Comment by wvangansbeke** [comment]
Hi @vobecant,

1. For clarity, our method also updates the decoder (i.e. deeplab head) during training for the semantic segmentation task. So, we can also use the head during evalution. This allows us to obtain feature maps that are learned for a particular downstream task and dataset. This is different from the standard MoCo setup. We used the evaluation as described in my previous post for MoCo. You could also remove our head during evaluation, but I don't see why that's a good idea. Our goal is to perform unsupervised semantic segmentation with standard semantic segmentation architectures.

2. You can check the network. We used a standard DeepLabv3. Should be a difference of a factor 8. You can upsample the output if you want the same resolution.

**Comment by vobecant** [comment]
Hi @wvangansbeke , thank you very much for your response.

Regarding the configurations for different initialization of linear evaluation (values reported in table 3a in the paper), you say that:

>  I want to point out that the optimal settings differ a lot between methods (e.g. SwAV, SeLa, SimCLR). For example SwAV uses a much larger learning rate for (linear) finetuning than MoCo. So there does not seem to be one set of parameters that works well for everything.

Based on this, could you please provide the configurations that you have used for the different methods?

Thank you very much in advance.

Best,
Antonin.

**Comment by wvangansbeke** [comment]
Hi @vobecant,

We use MoCo. All parameters are provided for MaskContrast and MoCo. If you want to use other pretraining strategies you'll have to do a grid search for the optimal parameters because they can differ between methods (e.g. InfoMin, SwAV, SeLa etc.). MoCo is still considered SOTA for general pretraining, so if you want to test other methods (e.g. InfoMin), just try some values. You can use a binary search strategy to find the optimal values quickly. The numbers in Table 3 can be used as guidance.

**Comment by wvangansbeke** [comment]
If there are more questions, I'm happy to answer them. Closing this issue for now.

**Comment by khangt1k25** [comment]
I am curious about training time of linear evaluation for the main model? 
I tried to finetune 1x1 Conv but it was slow, can you share your training time of your own config. Thank you
